packages/circle-sdk-setup/src/cli.ts

```diff
@@ -0,0 +1,202 @@
+
+  constructor(options: SetupOptions) {
+    this.options = {
+      envPath: path.resolve(process.cwd(), '.env'),
+      ...options,
+    };
+  }
+
+  async checkExistingConfig(): Promise<boolean> {
+    console.log('Checking existing configuration...');
+    try {
+      if (fs.existsSync(this.options.envPath)) {
+        const envContent = dotenv.parse(fs.readFileSync(this.options.envPath));
+        return !!(envContent.CIRCLE_SECRET && envContent.CIRCLE_APP_ID);
+      }
+      return false;
+    } catch (error) {
+      console.error('Error checking existing configuration:', error);
+      return false;
+    }
+  }
+
+  async generateSecret(): Promise<string> {
+    console.log('Generating secret...');
+    const secret = SecretApi.generateSecret();
+    console.log('Secret generated:', secret);
+    return secret;
+  }
+
+  async initializeSdk(apiKey: string, secret: string): Promise<CircleSdk> {
+    console.log('Initializing Circle SDK...');
+    return new CircleSdk(apiKey, secret);
+  }
+
+  async fetchConfiguration(sdk: CircleSdk) {
+    console.log('Fetching configuration...');
+    const config = await sdk.secret.getConfig();
+    console.log('Configuration fetched:', JSON.stringify(config, null, 2));
+    return config;
+  }
+
+  async fetchPublicKey(sdk: CircleSdk) {
+    console.log('Fetching public key...');
+    const publicKey = await sdk.secret.getPublicKey();
+    console.log('Public key retrieved');
+    return publicKey;
+  }
+
+  generateCiphertext(secret: string, publicKey: string) {
+    console.log('Generating entity secret ciphertext...');
+    const cipherText = SecretApi.getEntitySecretCiphertext(secret, publicKey);
+    console.log('Ciphertext generated');
+    return cipherText;
+  }
+
+  async registerCiphertext(sdk: CircleSdk, cipherText: string) {
+    try {
+      console.log('Registering entity secret ciphertext...');
+      const registration = await sdk.secret.registerEntitySecretCiphertext(cipherText);
+      console.log('Entity secret ciphertext registered');
+      return registration;
+    } catch (error: unknown) {
+      if (
+        (typeof error === 'object' && (error as any).code === -1) ||
+        ((error as Error).message &&
+          (error as Error).message.includes('Something went wrong'))
+      ) {
+        console.error('\n🚫 Error: Unable to register entity secret ciphertext.');
+        console.error('This might be due to the ciphertext already being registered.');
+        console.error('Please check your existing configuration in the Circle Console.');
+        process.exit(1);
+      }
+
+      // Re-throw other types of errors
+      throw error;
+    }
+  }
+
+  prepareEnvFile(config: any, secret: string, apiKey: string) {
+    console.log('Preparing .env file...');
+    let envContent = '';
+    if (fs.existsSync(this.options.envPath)) {
+      envContent = fs.readFileSync(this.options.envPath, 'utf-8');
+      // Remove existing Circle config if any
```

Comment:
Is that what we'd want to do in this situation? I guess this goes along with the comment I left regarding the `checkExistingConfig` function.